### PR TITLE
root: fix build on aarch64-linux

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

The error seems related to #201254, and the same fix works. Also #192581.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -36275,6 +36275,7 @@ with pkgs;
   root = callPackage ../applications/science/misc/root {
     python = python3;
     inherit (darwin.apple_sdk.frameworks) Cocoa CoreSymbolication OpenGL;
+    stdenv = if stdenv.isLinux then gcc11Stdenv else stdenv;
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix:36278]: Is this specific to aarch64-linux?

Comment (wegank) [reply pkgs/top-level/all-packages.nix:36278]: Yes. On x86_64-linux stdenv defaults to gcc11Stdenv, so there is no change.

Comment (veprbl) [reply pkgs/top-level/all-packages.nix:36278]: Okay, but what happens if x86 stdenv switches to gcc12, but aarch64 doesn't?

Comment (veprbl) [reply pkgs/top-level/all-packages.nix:36278]: Also if you could leave a comment on what error this fixes?

Comment (wegank) [reply pkgs/top-level/all-packages.nix:36278]: > Okay, but what happens if x86 stdenv switches to gcc12, but aarch64 doesn't?

Nice argument. Switched to aarch64-linux only and added comment.

**Review by veprbl — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -36275,6 +36275,8 @@ with pkgs;
   root = callPackage ../applications/science/misc/root {
     python = python3;
     inherit (darwin.apple_sdk.frameworks) Cocoa CoreSymbolication OpenGL;
+    # https://github.com/NixOS/nixpkgs/issues/201254
+    stdenv = if (stdenv.isLinux && stdenv.isAarch64) then gcc11Stdenv else stdenv;
```
Comment (veprbl) [inline pkgs/top-level/all-packages.nix:36279]:
```suggestion
    stdenv = if stdenv.isLinux && stdenv.isAarch64 && stdenv.cc.isGNU then gcc11Stdenv else stdenv;
```

**Pull request merged**